lib/PuppeteerSharp/Page.cs

```diff
@@ -160,6 +160,12 @@ private Page(Session client, Target target, FrameTree frameTree, bool ignoreHTTP
             return await context.EvaluateFunctionHandleAsync(pageFunction, args);
         }
 
+        public async Task EvaluateOnNewDocumentAsync(string pageFunction, params object[] args)
+        {
+            var source = Helper.EvaluationString(pageFunction, args);
```

Comment:
I think `EvaluationString` in Puppeteer tries to serialize a javascript function, I don't know if we need to do that.